pennylane/tape/tape.py

```diff
@@ -339,6 +358,31 @@ def interface(self):
         """str, None: automatic differentiation interface used by the quantum tape (if any)"""
         return None
 
+    @contextlib.contextmanager
+    def stop_recording(self):
```

Comment:
I think this context manager is good infrastructure, but I would like to also include a cleaner API for the user. It would be nice to have a `qml.stop_recording(fn)` transform that creates a new function that has the same arguments and output as `fn`, but just doesn't queue any ops.

You could add it reusing what you already have with something like this:


```python

def stop_recording(fn):
    @wraps(fn)
    def wrapper(*args, **kwargs):
        tape = qml.get_tape() or qml.QuantumTape()
        with tape.stop_record():
            return fn(*args, **kwargs)
    return wrapper
```

Reply:
Something like `op = qml.stop_recording(qml.RX)(0.123, wires=0)` would be a simple example. Another could be like:

```python

def list_of_ops(params):
  return [qml.RX(...), qml.RY(...), qml.RZ(...)]

@qml.qnode(dev):
def circuit(params):
  # Create the list of ops, but don't apply them.
  list_of_ops = qml.stop_recording(list_of_ops)(params)
```

Reply:
Have added this!